TransformAST.cpp

```diff
@@ -0,0 +1,142 @@
+#include <string>
+
+#include <swift/AST/ASTContext.h>
+#include <swift/AST/ASTWalker.h>
+#include <swift/AST/Expr.h>
+#include <swift/AST/ParameterList.h>
+#include <swift/AST/Pattern.h>
+#include <swift/AST/Stmt.h>
+
+#include "TransformAST.h"
+
+void TransformFinalExpressionAndAddGlobal(swift::SourceFile &src_file)
```

Comment:
Do you think it would make sense to turn this into two functions?

Reply:
I thought about it, but I think one function is better because these two things are coupled - we do the assignment if and only if we make the variable.

Reply:
Makes sense to me.